iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/service/metric/JvmGcMetrics.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.commons.service.metric;
+
+import org.apache.iotdb.commons.concurrent.IoTDBThreadPoolFactory;
+import org.apache.iotdb.commons.concurrent.threadpool.ScheduledExecutorUtil;
+import org.apache.iotdb.metrics.AbstractMetricService;
+import org.apache.iotdb.metrics.metricsets.IMetricSet;
+import org.apache.iotdb.metrics.utils.MetricLevel;
+
+import com.google.common.base.Preconditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class JvmGcMetrics implements IMetricSet {
+  private static final Logger logger = LoggerFactory.getLogger(JvmGcMetrics.class);
+  private final ScheduledExecutorService scheduledGCInfoMonitor =
+      IoTDBThreadPoolFactory.newSingleThreadScheduledExecutor("JVM-GC-Statistics-Monitor");
+  private Future<?> scheduledGcMonitorFuture;
+  // Ring buffers containing GC timings and timestamps when timings were taken
+  private final TsAndData[] gcDataBuf;
+  //    private final long maxGcTimePercentage = 0L;
+  private final long maxGcTimePercentage = 70L;
+  private final long observationWindowMs = TimeUnit.MINUTES.toMillis(1);
+  //        private final long observationWindowMs = TimeUnit.SECONDS.toMillis(1);
+  private final long sleepIntervalMs = TimeUnit.SECONDS.toMillis(5);
+  //        private final long sleepIntervalMs = TimeUnit.MILLISECONDS.toMillis(1000);
+  private final int bufSize;
+  private int startIdx;
+  private int endIdx;
+  private long startTime;
+  private final GcData curData = new GcData();
+  private boolean shouldRun = true;
+  private final GcTimeAlertHandler alertHandler;
```

Comment:
Consider remove some useless comment, and add description for each member variables